server/src/prompter.rs

```diff
@@ -0,0 +1,29 @@
+// org.gnome.keyring.internal.Prompter
+// https://gitlab.gnome.org/GNOME/gcr/-/blob/master/gcr/org.gnome.keyring.Prompter.xml
+
+use std::collections::HashMap;
+
+use oo7::dbus::ServiceError;
+use zbus::{
+    proxy,
+    zvariant::{OwnedObjectPath, OwnedValue},
+};
+
+#[proxy(
+    default_service = "org.gnome.keyring.SystemPrompter",
+    interface = "org.gnome.keyring.internal.Prompter",
+    default_path = "/org/gnome/keyring/Prompter"
+)]
+pub trait Prompter {
+    fn begin_prompting(&self, callback: OwnedObjectPath) -> Result<(), ServiceError>;
+
+    fn perform_prompt(
+        &self,
+        callback: OwnedObjectPath,
+        type_: &str, // 'password' or 'confirm', put this in Enum?
+        properties: HashMap<&str, OwnedValue>,
```

Comment:
I assume the properties are fixed/known dictionnaries, if so, they must be defined as structs.